### PR TITLE
15 combinelatest as function issue

### PR DESCRIPTION
**Pull request by LironHazan**

### CODE REVIEW

File: src/Mock.ts [hunk 1]
```diff
@@ -21,13 +20,15 @@ export class Mocker {
     private excludedPropertyNames: string[] = ["hasOwnProperty"];
     private defaultedPropertyNames: string[] = ["Symbol(Symbol.toPrimitive)", "then", "catch"];
 
-    constructor(private clazz: any, public instance: any = {}) {
+    constructor(private clazz: any, public instance: any = {}, isSpy: boolean = false) {
```
Comment (LironHazan) [inline src/Mock.ts:23]: I **hate** passing booleans as arguments (unless it's part of a configuration) but I wanted the process methods to not run when instantiating the Spy, I may refactor the class processing to use a parser in future

**Comment by johanblumenberg** [comment]
I think it works as a workaround, but long term I think it would be better to refactor the usage of Proxy a bit to make the code simpler instead of adding yet another special case.

I guess that the constructor code parsing is not needed on a spy, because the spy is attached to an existing object, and all dynamic method assignments are already done.

**Comment by LironHazan** [comment]
> I think it works as a workaround, but long term I think it would be better to refactor the usage of Proxy a bit to make the code simpler instead of adding yet another special case.
> 

Thanks, how would you suggest simplifying the Proxy code?
I'm not sure I have something in mind, I was thinking about getting the properties using Reflect.ownKeys(object) and iterate the result but it wont catch all cases..

**Comment by johanblumenberg** [comment]
> how would you suggest simplifying the Proxy code?

As I suggested here: https://github.com/TypeStrong/ts-mockito/pull/17

Right now there is one implementation for mocking classes, and a different implementation for mocking interfaces. This PR adds another difference between mocks and spies.

I think it would be better to simplify the code, and have one implementation that works the same for mocking classes and interfaces.

Compare this implementation here: https://github.com/TypeStrong/ts-mockito/blob/426058d3e84dcbc955b7b51843110ce0f9cc2af2/src/Mock.ts#L24
With the one in my fork: https://github.com/johanblumenberg/ts-mockito/blob/e5c009f9df45e4141744aaa92e4481735fd649b3/src/Mock.ts#L38

**Comment by LironHazan** [comment]
@johanblumenberg thanks, I'll need to deep dive a bit in that implementation (with the policies, I lake of context)
so meanwhile I'll merge this PR

**Pull request merged**